README.md

```diff
@@ -33,6 +35,29 @@ To begin configuring code insights, ensure that you have the necessary permissio
 - By following these steps, you'll have the `ruff2bitbucket` code insight configured in Bitbucket, streamline the code review process. This integration fosters a more collaborative and efficient development environment, ultimately leading to higher-quality software.
 
 ## Security
-When neither a username nor a password is explicitly provided (via the command line), the system will conduct a scan of environment variables for potential matches resembling `USER`/`USR` and `PASS`/`PW`.
+
+### Possibility 1 (username/password authentication):
+```ruff2bitbucket --user ... --pass ...```
+This will use the provided username and password to authenticate against BitBucket.
+
+### Possibility 2 (username/API token):
+```ruff2bitbucket --user ... --token ...```
+This will use the provided username and API token to authenticate against BitBucket.
+
+For more information on how to create this API token, please check [this link at Atlassian](https://support.atlassian.com/atlassian-account/docs/manage-api-tokens-for-your-atlassian-account/).
+
+### Possibility 3 (keyring authentication):
+```ruff2bitbucket --user ... --service_name ...```
+This will use the default keyring to retrieve the password from the provided service name for the provided user. This user and retrieved password will then be used to authenticate against BitBucket.
+
+For more information about [keyrings, click here](https://github.com/jaraco/keyring/blob/main/README.rst).
+
+### Possibility 4 (environment variables authentication):
+```ruff2bitbucket --user ... --passvar ...```
+This will use the contents of the environment variables as pointed to by `user` and `passvar` to authenticate against BitBucket.
+
+### Possibility 5 (automatic):
+```ruff2bitbucket```
+The script will conduct a scan of environment variables for potential matches resembling `USER`/`USR` and `PASS`/`PW`.
```

Comment:
Would a `--auto` flag to trigger this behaviour clash with your current uses?

Reply:
Actually yes. Because these cases all expect a "user" parameter.

So it's either: "user pass", "user token", "user service_name" or "user passvar".

I was trying to find the least effort to add these, and still keep the automated way as a fallback. (but if you use any of the other option, it won't fallback to the automated mode).